utils/configcheck.py

```diff
@@ -52,6 +52,31 @@ def sd_configcheck(agentConfig):
         configs = load_check_directory(agentConfig, get_hostname(agentConfig))
         get_sd_configcheck(agentConfig, configs)
 
+def docker_inspect():
+    print("-- docker_inspect -- \n")
+    dockerutil = DockerUtil()
+    containers = dockerutil.client.containers()
+    for cont in containers:
+        c_image = dockerutil.image_name_extractor(cont)
+        if 'datadog/docker-dd-agent' in c_image:
```

Comment:
That only works if the image is named like that.
- Some users replicate images they depend on in their self-hosted repository where docker-dd-agent may be named differently. We should check all RepoTags in this case.
- Trickier: if we can make this work even when users run a custom version of our image it would be best.

Reply:
@hkaj indeed, I figured I would do 2 loops, one for the image where we only inspect if we find it (as inspect is expensive) and then, if indeed the user is running a custom image then I [inspect all containers searching for the dd-agent env variable](https://github.com/DataDog/dd-agent/blob/5c6469d3a3128dfb6763a8c8702b19b0ffc2f5f6/utils/configcheck.py#L72-L79)
And I stop as we find one.

I can look into the RepoTags too.

Reply:
The flare command is run within the container. You can get the container id from within a container by parsing
`/proc/self/cgroup`

e.g. of output of `/proc/self/cgroup`

```
11:devices:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
10:freezer:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
9:perf_event:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
8:hugetlb:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
7:cpu,cpuacct:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
6:net_cls,net_prio:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
5:blkio:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
4:memory:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
3:cpuset:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
2:pids:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
1:name=systemd:/docker/928221b873eb753b220c6011c4cabb041771dd3508080c3cffe67989b1115b58
```